lib/utils/compose.js

```diff
@@ -519,7 +519,7 @@ const getPreviousRepos = (sdk, docker, logger, appID) =>
 		})
 		.then(function(release) {
 			// grab all images from the latest release, return all image locations in the registry
-			if (release.length > 0) {
+			if (release && release.length > 0) {
```

Comment:
The release && should be unnecessary as the line client should never return null for this

Reply:
> **should be** unnecessary [...] pine client **should** never

_vs._

> **is** unnecessary [...] pine client **will** never

That's OK, I am happy with the harmless redundancy when it gives me extra confidence. :-)

Reply:
If it ever gives you anything other than an array things have gone so badly wrong that bailing is actually the best you can hope for

Reply:
I would also say that should be doing
```
.then(function(([release])) {
```